### ISSUE TITLE
A BatchNorm inside a Scan never accumulates running statistics

### ISSUE THREAD

**Issue by jessegrabowski** [pymc-devs/pytensor-ml#125]
`collect_non_trainable_updates` walks `ancestors(outputs)` looking for a `StatefulOp` whose declared input *is* the `NonTrainableParameter` object. Inside a `Scan` that input is an inner nominal variable, and the walk does not enter `Scan.inner_outputs` in the first place, so a recurrent batch norm contributes no write-backs at all. Training runs, the loss falls, and the statistics sit at their initial values for the whole run.

```python
import numpy as np
import pytensor.tensor as pt
from pytensor.scan import scan

from pytensor_ml.layers import BatchNorm, Linear
from pytensor_ml.optim import adam
from pytensor_ml.pytensorf import (
    collect_non_trainable_updates,
    collect_trainable_params,
    function,
)

xseq = pt.tensor("xseq", shape=(None, None, 4))          # (time, batch, features)
lin, bn = Linear("l", 4, 4), BatchNorm("bn", 4)
out = scan(lambda xt: bn(lin(xt)), sequences=[xseq], return_updates=False)
loss = (out**2).mean()

print(list(collect_non_trainable_updates(loss)))   # []

step = function([xseq], loss, updates=adam(1e-2)(loss, collect_trainable_params(loss)))
X = np.random.default_rng(0).normal(size=(6, 8, 4)) + 50.0
for _ in range(20):
    step(X)
print(bn.running_mean.get_value())   # [0. 0. 0. 0.]  after 20 steps on data centered at 50
```
